### PR TITLE
Feature/deploy pip package

### PR DESCRIPTION
**Pull request by jamie-short**
Adding in the required files to allow creation of a pip package for the cli.

Following this PR, an Action will be setup on main to publish the pip package to pypi

### CODE REVIEW

File: docs/developer_notes.md [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
+The additional option `--hiddenimport=python-dateutil` is required, as pyinstaller fails to automatically pick up the codes dependency on the python-dateutil module.
+
+### Pip deployment
+The code has also been designed to be deployed as a pip package via pypi (https://pypi.org/user/dafni-facility/).
+
+If the code has been updated, and a new release is being made, the setup.py version number should be appropriately incremented, and then the following command run from the root directory of the repository:
+
+`python -m build`
+
+This will create a build and dist folder in the root folder, containing a `.tar.gz` and a `.whl` for the new version of the pip package ready to publish to pypi.
```
Comment (iroper-tessella) [inline docs/developer_notes.md:46]: Do you want to include how people can install locally? Or would it be better to only install from pypi?

**Review by iroper-tessella — COMMENTED** [review]
Just one question in the docs.
Additionally: Does this enable the automatic generation through a github action?

File: docs/developer_notes.md (hunk 1 above)
Comment (jamie-short) [reply docs/developer_notes.md:46]: once its up on pypi, it ill be a matter of `pip install dafni-cli`

Comment (jamie-short) [reply docs/developer_notes.md:46]: As for github action, it will need setting up, but it looks like a 5-10 minute task to do so

**Review by iroper-tessella — APPROVED** [review]
Questions answered and it looks good.

**Pull request merged**